holoviews/core/data/iris.py

```diff
@@ -280,12 +285,18 @@ def select_to_constraint(cls, dataset, selection):
         """
         Transform a selection dictionary to an iris Constraint.
         """
+        def get_slicer(start, end):
+            def slicer(cell):
+                return start <= cell.point < end
+            return slicer
         constraint_kwargs = {}
         for dim, constraint in selection.items():
             if isinstance(constraint, slice):
                 constraint = (constraint.start, constraint.stop)
             if isinstance(constraint, tuple):
-                constraint = iris.util.between(*constraint, rh_inclusive=False)
+                if constraint == (None, None):
+                    continue
+                constraint = get_slicer(*constraint)
```

Comment:
I don't know what happened here, I think it's an iris bug, but I'm not bothered to investigate tbh. The iris interface can move to geoviews in 2.0.